Graphql Codegen to create hooks and functions directly from schema

SlayerAnsh opened the pull request:
# Motivation

Updates to GQL queries required adding document nodes and query typings for variable and result. This new package will help auto generate the typings and craete helper functions and hooks which can directly be integrated in projects.

# Package Features

- Generate variables and result types directly from schema.
- Validate Document Nodes against remote schema so there is very less chance for typo error
- Fragments can be easily used for the queries
- Ready to use integration for graphql-request and apollo-client hooks

EDIT
- Now codegen auto generate all the operations for you. If you need a simple query, codegen probably has it generated already. Only when you need a special query, for example, getting app components and config in single query, you can write custom schema files. But if you needed only config or only components, these schema are autogenerated. Even all the arguments based fields like `isOperator` has a dedicated schema auto generated for it.
- Scripts and folder structure has been updated
- Added new types.ts file which contain all the base types used in gql like AndrSearch, TxInfo, etc.

# Testing

Tested integration for app and base ado query in cli and webapp locally

# Future work

(Done) ~~We need to port existing queries here. This can be done in incremental way as demonstrated with app query.~~

SlayerAnsh committed:
Added gql codegene and few queries sample

SlayerAnsh committed:
Added readme and updated codegen and type policy

SlayerAnsh committed:
Added new queries needed for webapp

SlayerAnsh committed:
Updated assets query key fields

SlayerAnsh committed:
More powerful codegen to auto generate all queries directly from schema

SlayerAnsh committed:
Codegen update and andrjs gql updates

SlayerAnsh committed:
Added support for subscription queries

arun567 reviewed (approved):
Code changes look fine.

leifasorensen reviewed (commented):
### Review
Code submission looks ready for approval after two minor adjustments:
- package.json scripts reference undeclared dependencies (e.g. jest, rimraf), and should be adjusted prior to package release.
- README.md contains some minor typographical and grammatical issues. Adjustments are recommended.

SlayerAnsh committed:
Fixed unused dependencies

leifasorensen committed:
Update README.md

Minor grammatical, spelling, and explanative adjustments to the codegen README file.
